IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m

```diff
@@ -93,6 +97,33 @@ - (MSIDDeviceMode)deviceModeEnumFromString:(NSString *)deviceModeString
     return MSIDDeviceModePersonal;
 }
 
+- (NSString *)ssoExtensionModeStringFromEnum:(MSIDSSOExtensionMode)ssoExtensionMode
+{
+    switch (ssoExtensionMode) {
+        case MSIDSSOExtensionModeFull:
+            return @"full";
+        case MSIDSSOExtensionModeSilentOnly:
+            return @"silent_only";
+        default:
+            return nil;
+    }
+}
+
+- (MSIDSSOExtensionMode)ssoExtensionModeEnumFromString:(NSString *)ssoExtensionModeString
```

Comment:
nit: this method is supposed to convert enum to string. I'd prefer to keep it just about it. The logic that determines which mode to use on which iOS version should preferably be outside of it separately.

Reply:
agree :)